fvm/state/state_transaction.go

```diff
@@ -22,13 +22,9 @@ type nestedTransactionStackFrame struct {
 	parseRestriction *common.AddressLocation
 }
 
-// TODO(patrick): rename to StateTransaction
-// StateHolder provides active states
-// and facilitates common state management operations
-// in order to make services such as accounts not worry about
-// the state it is recommended that such services wraps
-// a state manager instead of a state itself.
-type StateHolder struct {
+// StateTransaction provides active transaction states and facilitates common
+// state management operations.
+type StateTransaction struct {
```

Comment:
I like the rename, but just want to throw another alternative out in case you like it better: `TransactionState`

Reply:
sure.  will rename